tests/test_raster.py

```diff
@@ -109,6 +109,17 @@ def test_arr(self):
         self.assertEqual(img_last.arr.shape, (385, 502, 1))
         img_last.close()
 
+    def test_arr_nodata(self):
+        array = np.ones((1, 385, 502))
+        array[0, 0, 0] = 0
+
+        img_nodata = Image(
+            array, crs=self.img.dataset.crs, transform=self.img.dataset.transform, nodata=0
+        )
+
+        self.assertEqual(img_nodata.dataset.nodata, 0.0)

```

Comment:
Why are you mixing float and int? You also don't even have to have nodata in an array to set it, right?
Please also check with `nodatavals`.

Reply:
I did not check with `nodatavals` as I think it requires a separate discussions on how to treat nodata values across different bands.

Reply:
You are right: it is not required to have the actual nodata value in the array. However, I included it for the sake of completeness.

Reply:
> Why are you mixing float and int?
The returned nodata value is of type float and it was intuitive for me to test for float but rasterio does not care about the datatype of the number. Changed as suggested by you.